cycler.py

```diff
@@ -335,7 +335,11 @@ def __repr__(self):
     def _repr_html_(self):
         # an table showing the value of each key through a full cycle
         output = "<table>"
-        sorted_keys = sorted(self.keys)
+        try:
+            sorted_keys = sorted(self.keys)
```

Comment:
The iteration order should be fixed for a given set in a given processes.  IIRC the iteration order is based on the internal hash of the key.

```python
ss = set("this is a crazy test")
lss = list(ss)
assert all(map(lambda x: x[0] == x[1], zip(lss, ss)))
```

seems to work.

I am 20/80 on the repr fallback,  The intent of this is to make the ordering stable run-to-run, but if you fall back to repr which may include a memory address which will be random anyway and the repr call is just over head.

On the other hand, this code path only happen when you display the cycler in a IPython notebook so the overhead does not matter and it _does_ do the right thing if your reprs are good.